shared/js/background/tracker-utils.js

```diff
@@ -30,6 +31,114 @@ function isTracker (url) {
     return !!tracker
 }
 
+/**
+ * Determine if the social entity should be blockedon this URL. returns True if so.
+ */
+function shouldBlockSocialNetwork (entity, url) {
+    const domain = tldts.parse(url).domain
+    const excludeData = getDomainsToExludeByNetwork()
+    return excludeData.filter(e => e.domain === domain && e.entity === entity).length === 0
+}
+
+/*
+ * Parse the social config to find excluded domains by social tracker. This then returns a list of objects
+ * that include the exlcuded domain and network, for use in other exception list handling.
+ */
+function getDomainsToExludeByNetwork () {
+    const excludes = []
+    for (const [entity, data] of Object.entries(tdsStorage.ClickToLoadConfig)) {
+        if (data.excludedDomains) {
+            const excludedDomains = data.excludedDomains.map(e => e.domain)
+            for (const domain of excludedDomains) {
+                excludes.push({
+                    entity: entity,
+                    domain: domain
+                })
+            }
+        }
+    }
+    return excludes
+}
+
+// Return true if URL is in our click to load tracker list
+function getSocialTracker (url) {
+    if (!facebookExperimentIsActive()) {
+        return
+    }
+    const parsedDomain = tldts.parse(url)
+    for (const [entity, data] of Object.entries(tdsStorage.ClickToLoadConfig)) {
```

Comment:
How many entries are in this config, and how large do we expect it to become? Does it make sense to build a lookup table to get O(1) lookups instead of O(N)? i.e. use an object with the keys being domains and the values being `data` from the `ClickToLoadConfig`?

Reply:
It's a great idea, but I'm not sure it's worth it here. There is only going to be one entry per social site, so N > 20 or so is unlikely (right now, N=1). Each entry can have multiple domain/subdomains included or excluded, so the extra complexity doesn't seem worth it here to me, though happy to discuss more.